server/src/main/java/org/opensearch/cluster/metadata/IndexMetadata.java

```diff
@@ -321,9 +321,11 @@ public Iterator<Setting<?>> settings() {
 
     public static final String SETTING_REMOTE_STORE_ENABLED = "index.remote_store.enabled";
 
-    public static final String SETTING_REMOTE_SEGMENT_STORE_REPOSITORY = "index.remote_store.segment.repository";
+    public static final String SETTING_REMOTE_SEGMENT_STORE_DATA_REPOSITORY = "index.remote_store.segment.repository";
```

Comment:
Should we add a setting for data repository as well? For validation purposes also it can simplify things - either the existing setting is there or the 2 new settings are there.

Reply:
Trying to avoid too many settings. Validation wise, the existing implementation is also keeping things simple. With one more setting, we need to handle cases where both the settings are provided.